### ISSUE TITLE
useQuery object format and queryKey override

### ISSUE THREAD

**Issue by ivan-dalmet** [ts-rest/ts-rest#297]
Hello the ts-rest team! Thanks a lot for the tool!

I have a little issue coming from the latest version of react query and trying to migrate a project to ts-rest (client only for now).

In latest version of react query we prefer a lot to use the object format instead of the params format for the useQuery.

```ts
// 😕 Params syntaxe
useQuery(queryKey, queryFn, queryOptions);

// ✅ Object syntaxe
useQuery({ queryKey, queryFn, ...queryOptions });
```

We prefer the object syntaxe because it's A LOT more simple to type query options with custom hooks but also because it's very more simple to override the queryKey. Typing with ts-rest is not the issue but overriding the queryKey is.

You can see the problem on this codesandbox https://codesandbox.io/p/sandbox/ts-rest-react-query-q2x534?file=/app/page.tsx:35,66

But in TL;DR 👇

```ts
// 😕 Current syntaxe 
// `queryKey` in `options` WILL NOT override the `queryKey`
client.getSomething.useQuery(["something"], params, options);

// 💡 Wanted syntaxe that can match the react query object syntaxe
// `queryKey` in `options` WILL override the `queryKey`
client.getSomething.useQuery({
  queryKey: ["something"],
  params,
  ...options,
});
```

I'm not sure if I'm 100% clear, feel free to ask me anything 😅

**Comment by Gabrola** [comment]
@ivan-dalmet thank you for the suggestion and the code sample! Quick question, does the the query key override if I use the parameter syntax in regular react-query? So `useQuery(['foo'], async () => fetch(..), { queryKey: ['override-key'] })`, does this work? Because if it does, it _should_ work in ts-rest because we pass the entire options object to the `useQuery` function.

Also yeah, it makes sense to omit the `queryFn` from `UseQueryOptions`, will do that!

**Comment by ivan-dalmet** [comment]
No it's no working in regular react query, that why object syntax is preferred in regular react query.

**Comment by Talent30** [comment]
Hi, @Gabrola. 

Is a recommended practice to using object as useQuery option  according to https://github.com/TanStack/query/blob/main/packages/eslint-plugin-query/src/rules/prefer-query-object-syntax

Addtionally is going to be the only way in react-query v5  according to https://tanstack.com/query/v5/docs/react/guides/migrating-to-v5

I am planning to send out a PR if you are happy with it.

**Comment by scarf005** [comment]
Hi @Gabrola, could we learn the reason this issue is closed as not planned? [react-query is V5](https://www.npmjs.com/package/@tanstack/react-query) which means ts-rest uses legacy parameters format.

**Comment by Gabrola** [comment]
@scarf005 closed in favor of #415. We are already working on the migration to v5, but it will have to be a part of a major version release of ts-rest v4 in which we are planning to introduce many changes, hence why the reason for the delay
